crates/generate/src/types/enum.rs

```diff
@@ -77,37 +77,25 @@ pub(super) fn define_enum(names: &Names, name: &witx::Id, e: &witx::EnumDatatype
         }
 
         impl<'a> wiggle_runtime::GuestType<'a> for #ident {
-            fn size() -> u32 {
-                ::std::mem::size_of::<#repr>() as u32
+            fn guest_size() -> u32 {
+                #repr::guest_size()
             }
 
-            fn align() -> u32 {
-                ::std::mem::align_of::<#repr>() as u32
-            }
-
-            fn name() -> String {
-                stringify!(#ident).to_owned()
-            }
-
-            fn validate(location: &wiggle_runtime::GuestPtr<'a, #ident>) -> Result<(), wiggle_runtime::GuestError> {
-                use ::std::convert::TryFrom;
-                let raw: #repr = unsafe { (location.as_raw() as *const #repr).read() };
-                let _ = #ident::try_from(raw)?;
-                Ok(())
```

Comment:
Could you verify that I'm reading it right? So now `validate` is effectively done as part of the `read` call?

Reply:
And as an added bonus, we also validate `write`s, right?

Reply:
Right yeah, reading implicitly validates as it goes along, with the main validation happening for enums/flags. (and the implicit validation of all pointers everywhere too)

Reply:
Oh, also, pointer validation happens on writes but data member validation doesn't, we always just write out what the bits happen to be

Reply:
Yep, awesome! This looks really good!